Update server-3-install-migration.adoc (#5355)
circleci/circleci-docs: jekyll/_cci2/server-3-install-migration.adoc

:toc: macro
:toc-title:

> Migrating from 2.19.x to 3.0 requires backing up your 2.19 instance data (Mongo, Postgres, and Vault) and than restoring that data in a waiting Server 3.0

Reviewer: ```suggestion
Migrating from 2.19.x to 3.0 requires you to back up your 2.19 instance data (Mongo, Postgres, and Vault) and then restore that data in a waiting Server 3.0
```

> instance. If you run into trouble you can fallback to your 2.19 instance. It does require an already operating Server 3.0 installation. Depending 

Reviewer: ```suggestion
instance. If you run into trouble you can fallback to your 2.19 instance, in this case your 2.19 installation will need to be restarted. Before starting the migration process you will need to have an operational Server 3.x installation up and running. Depending 
```

Reviewer: @nathanfishcircleci could you check if this works?

Reviewer 2: Customers can start the application with `replicatedctl app start` or from the UI of the management console.

> instance. If you run into trouble you are able to fallback to your 2.19 instance. It does require an already operating Server 3.0 installation. Depending 

Reviewer: ```suggestion
instance. If you run into trouble you can fallback to your 2.19 instance. It does require an already operating Server 3.0 installation. Depending 
```

> instance. If you run into trouble you are able to fallback to your 2.19 instance. It does require an already operating Server 3.0 installation. Depending 

Reviewer: I'm not clear on what "It does require an already operating Server 3.0 installation" means here - the fallback process doesn't require an operational 3.0 installation?

Reviewer: Oh I see, that just means an operational 3.0 install is required before starting the migration.

> instance. If you run into trouble you are able to fallback to your 2.19 instance. It does require an already operating Server 3.0 installation. Depending 

Reviewer 2: Might be worth calling out the 2.19.x application will be stopped; in the event they need to rollback they will need to start the application back up.

Author: @ryanwohara sounds like a great addition.

Reviewer: I've added this in my change below, but maybe we need to call out how to restart 2.19 in this event?

The migration script will:

* Stop your v2.19.x application

> * Perform pre-flight checks to confirm namespace and datastores for 2.19.x. 

Reviewer: ```suggestion
* Perform pre-flight checks to confirm namespace and datastores for 2.19.x
```

> * Create a tarball of your v2.19.x application's PostgreSQL and Mongo databases
> * Archive existing application data for Vault and CircleCI encryption/signing keys

Reviewer: ```suggestion
* Create a tarball of your v2.19.x application's PostgreSQL and Mongo databases.
* Archive existing application data for Vault and CircleCI encryption/signing keys.
```

> * Export the 2.19.x tarball to your v3.x installation (exported data stores are stored in `migrate/circleci_export`, which may be useful for debugging purposes).

Reviewer 2: We should call out here we are actually copying the files locally before initiating the 3.x import.

Reviewer 2: Also `migrate/circleci_export` is slightly incorrect - the `circleci_export` directory will always be in the current working directory. If a user were to `cd /tmp` before running the migration script, the export would end up in `/tmp/circleci_export`.

> * Export the 2.19.x tarball to your v3.x installation (exported data stores are stored in `migrate/circleci_export`, which may be useful for debugging purposes).

Reviewer: ```suggestion
* Export the 2.19.x tarball to your v3.x installation. Exported data stores are stored in a directory named `circleci_export`, located relative to wherever the migration script is run from. This can be useful for debugging purposes.
```

Reviewer: @ryanwohara is this better?

Reviewer 2: Love it :)

### Why does nothing happen when I select "Start Building" on my project after migration?
* By default, a newly added project (a project that has never been followed) will trigger a build automatically after it has been followed for the first time. If the project was or ever has been followed in 2.0 or 3.0, it will not be considered a new project or first build and a build will not be triggered after follow. To trigger a build, perform an activity that will trigger a Github webhook such as pushing up a new commit or branch.

### I got an error "Error from server (NotFound):" 

> * The script assumes a specific naming patterns for your Postgres and MongoDB. If you get this error, it may indicate a non-standard installation, a missing DB migration, or other issues. In this case it is best to contact support with a support bundle and the output from the migration script. 

Reviewer: ```suggestion
* The script assumes specific naming patterns for your Postgres and MongoDB. If you get this error, it may indicate a non-standard installation, a missing DB migration, or other issues. In this case it is best to contact support with a support bundle and the output from the migration script. 
```
